### PR TITLE
Build scripts, CI scripts and Server Plugin

### PR DESCRIPTION
**Pull request by loosebazooka**

### CODE REVIEW

**Comment by loosebazooka** [comment]
@patflynn @meltsufin @tangiel

The first half of the plugin is the server side with two tasks

```
endpointsClientLibs
endpointsDiscoveryDocs
```

and provides one artifact and accompanying configuration for dependent projects to extract

```
discover docs - an archive of all discovery docs created by this server
```

File: src/main/java/com/google/cloud/tools/gradle/endpoints/server/task/GenerateClientLibsTask.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Endpoints task to download a client library document from the endpoints service, useful for
+ * projects not in the same gradle configuration. For clients and servers that are part of the
+ * same gradle project, use EndpointsClientPlugin
+ */
+public class GenerateClientLibsTask extends DefaultTask {
```
Comment (patflynn) [inline src/main/java/com/google/cloud/tools/gradle/endpoints/server/task/GenerateClientLibsTask.java:40]: Is this generating a jar or sources?

File: src/main/java/com/google/cloud/tools/gradle/endpoints/server/task/GenerateClientLibsTask.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import org.gradle.api.tasks.TaskAction;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Endpoints task to download a client library document from the endpoints service, useful for
+ * projects not in the same gradle configuration. For clients and servers that are part of the
+ * same gradle project, use EndpointsClientPlugin
```
Comment (patflynn) [inline src/main/java/com/google/cloud/tools/gradle/endpoints/server/task/GenerateClientLibsTask.java:38]: period at end of sentence.

**Comment by loosebazooka** [comment]
closed by #3

**Pull request closed without merging**